src/Tools/dotnet-monitor/StreamExtensions.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal static class StreamExtensions
+    {
+        private const int DefaultBufferSize = 4096;
```

Comment:
Did you want to match the default FileStream buffer size here (4KiB, which coincides with the default page size)? Also consider being explicit about the FileStream buffer size.

Reply:
The default buffer size for FileStream is 4096: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs#L15